tests/shell/src/main.c

```diff
@@ -0,0 +1,132 @@
+/** @file
+ *  @brief Interactive shell test suite
+ *
+ */
+
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+
+#include <shell/shell.h>
+
+static void test_shell_exec(const char *line, int result)
+{
+	char cmd[80];
+	int ret;
+
+	strncpy(cmd, line, sizeof(cmd));
+	ret = shell_exec(cmd);
+
+	TC_PRINT("shell_exec(%s): %d\n", line, ret);
+
+	zassert_true(ret == result, line);
+}
+
+static void test_help(void)
+{
+	TC_PRINT("\nhelp tests:\n");
```

Comment:
this duplicates what the testsuite already prints, too verbose. I would remove all those prints, thats how it looks now:

`tc_start() - test_select

select tests:`